packages/core/src/utils/LockManager.ts

```diff
@@ -0,0 +1,47 @@
+import { v4 as uuidv4 } from 'uuid';
+
+class LockManager {
+  private static _instance: LockManager;
+  private locks: Map<string, boolean>;
+
+  private constructor() {
+    this.locks = new Map();
+  }
+
+  public static get instance() {
+    if (!LockManager._instance) {
+      LockManager._instance = new LockManager();
+    }
+
+    return LockManager._instance;
+  }
+
+  createLock(): string {
```

Comment:
Would it be more flexible to allow passing in a lockId, and only generate one if not provided? What do you think?

Reply:
I've made updates to the function to accept an optional `lockId`. This allows for more flexible operations.